File: solutions/csharp/grains/1/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64)
        {
            throw new ArgumentOutOfRangeException("Square must be larger than 0 and smaller or equal to 64.");
        }
        return (ulong)Math.Pow(2, n - 1);
    }

    public static ulong Total() => ulong.MaxValue;
}

File: solutions/python/grains/1/grains.py
"""
Functions for calculating the number of grains on a chessboard.
"""

def square(number):
    if number > 0 and number < 65:
        return 2 ** (number - 1)
    else:
        raise ValueError("square must be between 1 and 64")

def total():
    return sum(map(lambda x: 2 ** x, [*range(0, 64)]))
